cli/lsp/diagnostics.rs

```diff
@@ -646,13 +645,12 @@ async fn update_diagnostics(
         error!("Error generating lint diagnostics: {}", err);
       })
       .unwrap_or_default();
-    {
-      let mut collection = collection.lock().await;
-      for diagnostic_record in diagnostics {
-        collection.set(DiagnosticSource::DenoLint, diagnostic_record);
-      }
```

Comment:
Race condition was right here where the lock is released and then the lock was acquired in `publish_diagnostics`.